extensions/cornerstone/src/init.js

```diff
@@ -275,3 +287,228 @@ const _connectToolsToMeasurementService = measurementService => {
+        const targetNormal = targetImagePlane.rowCosines
+          .clone()
+          .cross(targetImagePlane.columnCosines);
+        const referenceNormal = referenceImagePlane.rowCosines
+          .clone()
+          .cross(referenceImagePlane.columnCosines);
+        let angleInRadians = targetNormal.angleTo(referenceNormal);
+        angleInRadians = Math.abs(angleInRadians);
+        if (angleInRadians < 0.5) {
+          console.error(
+            'Could not render reference lines, angle in radians does not match the expected value'
+          );
+          return;
+        }
+
+        const points = planeIntersection(targetImagePlane, referenceImagePlane);
+
+        if (!points) {
+          console.error('Could not render reference lines, points not defined');
+          return;
+        }
+
+        const referenceLine = {
+          start: projectPatientPointToImagePlane(
+            points.start,
+            referenceImagePlane
+          ),
+          end: projectPatientPointToImagePlane(points.end, referenceImagePlane),
+        };
+
+        if (!referenceLine) {
+          console.error(
+            'Could not render reference lines, no reference line to render'
+          );
+          return;
+        }
+
+        const onReferenceElementImageRendered = () => {
+          const context = getNewContext(referenceElement.canvas);
+          context.setTransform(1, 0, 0, 1, 0, 0);
+          draw(context, newContext => {
+            drawLine(
+              newContext,
+              referenceElement.element,
+              referenceLine.start,
+              referenceLine.end,
+              { color: 'greenyellow' }
+            );
+          });
+
+          referenceElement.element.removeEventListener(
+            cornerstone.EVENTS.IMAGE_RENDERED,
+            onReferenceElementImageRendered
+          );
+        };
+
+        referenceElement.element.addEventListener(
+          cornerstone.EVENTS.IMAGE_RENDERED,
+          onReferenceElementImageRendered
+        );
+        cornerstone.updateImage(referenceElement.element);
+      });
+  };
+
+  cornerstone.events.addEventListener(
+    cornerstone.EVENTS.ELEMENT_DISABLED,
+    event => {
+      if (event.detail && event.detail.element) {
+        event.detail.element.removeEventListener(
+          cornerstone.EVENTS.IMAGE_RENDERED,
+          renderReferenceLines
+        );
+      }
+    }
+  );
+
+  const bindEnabledElementsEventListeners = enabledElements => {
+    enabledElements.forEach(enabledElement => {
+      enabledElement.addEventListener(
+        cornerstone.EVENTS.IMAGE_RENDERED,
+        renderReferenceLines
+      );
+    });
+  };
+
+  const unbindEnabledElementsEventListeners = enabledElements => {
+    enabledElements.forEach(enabledElement => {
+      enabledElement.removeEventListener(
+        cornerstone.EVENTS.IMAGE_RENDERED,
+        renderReferenceLines
+      );
+    });
+  };
+
+  let previousLayout;
+  window.store.subscribe(() => {
```

Comment:
Do we need any of this? Can't we just use `ELEMENT_ENABLED` for add and `ELEMENT_DISABLED` for remove? `          csTools.store.state.enabledElements` would being the source of truth for sync targets

Reply:
I remember I had some problems with the approach you described, but it seems to be working fine now.